notebooks/introductory_infilling.ipynb

```diff
@@ -0,0 +1,2572 @@
+      "\n",
+      "        **kwargs\n",
+      "            Keyword arguments used by this class to derive the relationship between\n",
+      "            ``variable_follower`` and ``variable_leaders``.\n",
+      "\n",
+      "        Returns\n",
+      "        -------\n",
+      "        :obj:`func`\n",
+      "            Function which takes a :obj:`pyam.IamDataFrame` containing\n",
+      "            ``variable_leaders`` timeseries and returns timeseries for\n",
+      "            ``variable_follower`` based on the derived relationship between the two.\n",
+      "            Please see the source code for the exact definition (and docstring) of the\n",
+      "            returned function.\n",
+      "        \n"
+     ]
+    }
+   ],
+   "source": [
+    "print(silicone.database_crunchers.base._DatabaseCruncher.derive_relationship.__doc__)\n",
+    "# # an alternative, pop-up interface is shown if you uncomment the line below\n",
+    "#silicone.database_crunchers.base._DatabaseCruncher.derive_relationship?"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "These crunchers are best explored by looking at the examples below.\n",
+    "\n",
+    "### Closest Root Mean Square cruncher\n",
+    "\n",
+    "This cruncher finds the scenario and model in the infiller database that minimises the RMS distance between the the lead emissions in the infiller and infillee data. The follow data is then directly taken from that scenario. \n",
+    "The full documentation is as follows:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "\n",
+      "    Database cruncher which uses the 'closest RMS' technkque.\n",
+      "\n",
+      "    This cruncher derives the relationship between two variables by finding the\n",
+      "    scenario which has the closest lead gas timeseries in the database.  The follower\n",
+      "    gas timeseries is then simply copied from the closest scenario.\n",
+      "\n",
+      "    Here, 'closest' is defined as the smallest time-averaged root mean squared (L2)\n",
+      "    difference.\n",
+      "\n",
+      "    .. math::\n",
+      "        RMS = \\left ( \\frac{1}{n} \\sum_{t=0}^n (E_l(t) - E_l^{d}(t))^2 \\right )^{1/2}\n",
+      "\n",
+      "    where :math:`n` is the total number of timesteps in the lead gas' timeseries,\n",
+      "    :math:`E_l(t)` is the lead gas emissions timeseries and :math:`E_l^d(t)` is a lead\n",
+      "    gas emissions timeseries in the database.\n",
+      "    \n"
+     ]
+    }
+   ],
+   "source": [
+    "print(silicone.database_crunchers.DatabaseCruncherRMSClosest.__doc__)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Infilling\n",
+    "\n",
+    "Firstly, let's cut the database down to a size that is comprehensible."
```

Comment:
It would probably be good to also print (a snapshot) of the reduced database here, e.g. by using `.timeseries()`, or otherwise `.scenarios()`, `.head()`, `.variables()`, or `.models()`.